TTS/enhancer/layers/losses.py

```diff
@@ -0,0 +1,91 @@
+from typing import Type
+import torch
+from TTS.utils.audio import TorchSTFT
+from TTS.tts.layers.losses import L1LossMasked
+from TTS.vocoder.layers.losses import MelganFeatureLoss, MSEGLoss, MSEDLoss, _apply_G_adv_loss, _apply_D_loss
+
+class BWEGeneratorLoss(torch.nn.Module):
+    def __init__(self, device):
+        super().__init__()
+        self.l1_masked= L1LossMasked(False)
+        self.specs = [TorchSTFT(
+                        n_fft=512*2**i, 
+                        hop_length=int(512*2**i/4),
+                        win_length=512*2**i,
+                        sample_rate=48000,
+                        device=device,
+                        do_amp_to_db=True) for i in range(4)]
+        self.mel_spec = TorchSTFT(
+                            n_fft=1024, 
+                            hop_length=512, 
+                            win_length=1024, 
+                            sample_rate=48000, 
+                            n_mels=128,
+                            use_mel=True,
+                            device=device,
+                            do_amp_to_db=True)
+        self.feat_match_loss = MelganFeatureLoss()
+        self.mse_loss = MSEGLoss()
+
+    def forward(self, y_hat, y, lens, scores_fake=None, feats_fake=None, feats_real=None):
+        return_dict = {}
+
+        # Waveform loss
+        return_dict["l1_wavform"] = self.l1_masked(y_hat, y, lens)
+        return_dict["loss"] = return_dict["l1_wavform"] * 50
+
+        # Compute spectrograms
+        y_specs = [self.specs[i](y[:, 0, :]) for i in range(4)]
+        y_mel = self.mel_spec(y[:, 0, :])
+
+        y_hat_specs = [self.specs[i](y_hat[:, 0, :]) for i in range(4)]
+        y_hat_mel = self.mel_spec(y_hat[:, 0, :])
+
+        # Spectrogram loss
+        mel_lens = self.compute_lens(y_mel, lens)
+        return_dict["l1_mel"] = self.l1_masked(y_hat_mel, y_mel, mel_lens)
+        return_dict["loss"] = return_dict["l1_mel"]
```

Comment:
You need to add with += not with = otherwise the l1_wavform loss will be ignored.

Reply:
Good catch!